### PR TITLE
[FEATURE] Added trait for data helper functionalityg

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: Traits/DataHelperTrait.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @return mixed
+     */
+    private function getProperty(array $data, $key, $default = null)
+    {
+        return array_key_exists($key, $data) ? $data[$key] : $default;
+    }
+
+    /**
+     * Sets a date if it's set in data.
```
Comment (notsure) [inline Traits/DataHelperTrait.php]: its

File: Traits/DataHelperTrait.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     *
+     * @param array $data
+     * @param string $key
+     * @param string|\DateTime $defaultDate
+     * @param callable $setCallback
+     */
+    private function setDate($data, $key, $defaultDate, callable $setCallback)
+    {
+        if (($date = $this->getProperty($data, $key, $defaultDate)) !== null) {
```
Comment (notsure) [inline Traits/DataHelperTrait.php]: declare variable outside of if

File: Traits/DataHelperTrait.php (hunk 1 above)
Comment (turbo-ele) [inline Traits/DataHelperTrait.php]: well, no

Comment (notsure) [inline Traits/DataHelperTrait.php]: it is

**Comment by notsure** [comment]
ack

**Pull request merged**